### ISSUE TITLE
bug: capacitor errors reading response headers android

### ISSUE THREAD

**Issue by tbscode** [ionic-team/capacitor#6604]
### Capacitor Version
```
Latest Dependencies:

  @capacitor/cli: 5.0.3
  @capacitor/core: 5.0.3
  @capacitor/android: 5.0.3
  @capacitor/ios: 5.0.3

Installed Dependencies:

  @capacitor/cli: 5.0.2
  @capacitor/android: 5.0.2
  @capacitor/ios: 5.0.2
  @capacitor/core: 5.0.2
```

### Platform(s)

- android

not tested on ios

### Current Behavior

I am encountering an issue with capacitor-http and the response headers for fetch calls on Android. My fetch calls are working fine in the web app, but they throw a Capacitor error when running on Android.

```
(index):392 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'startsWith')
    at window.fetch ((index):392:80) <------- error in capacitor android
    at async g (index-e974cbdf3f503cbf.js:1:2107)
```

The line causing the error seems to be this one: https://github.com/ionic-team/capacitor/blob/6843d9642fad9a322579cbe5f01563929a83dbf5/android/capacitor/src/main/assets/native-bridge.js#L387

## In the browser

I am using Next.js, React, and Capacitor for the frontend of a cross-platform React app. The backend is a Django server set up with the REST framework.

For the frontend of the browser web application, I can use the following code:

```js
const res = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/user_data`, {
  method: 'POST',
  credentials: 'include',
  headers: {
    'X-CSRFToken': getCookiesAsObject().csrftoken,
    'Content-Type': 'application/json',
    Accept: 'application/json'
  },
  body: JSON.stringify({'some' : 'data'}),
});
```

When I run the web app, I get the expected `200 OK` response with some user data.

These are the server headers:

```
HTTP/1.1 200 OK
date: Wed, 17 May 2023 21:37:55 GMT
server: uvicorn
content-type: application/json
vary: Accept, origin, Cookie
allow: POST, GET, OPTIONS
access-control-allow-credentials: true
access-control-allow-origin: http://localhost:3000
x-frame-options: DENY
content-length: 95
x-content-type-options: nosniff
referrer-policy: same-origin
cross-origin-opener-policy: same-origin
```

### On Android:

I modify `NEXT_PUBLIC_BACKEND_URL = ...`, and the app builds and starts correctly on Android. I see all requests coming through correctly in the logs of my Django server, data is corretly returned and response headers are set. 

Using middleware I've checked my headers are set correctly, including the CSRF token and session ID, also making a call to my login api succeeds and capacitor correctly sets cookies, but then fails with the same error.

It seems like Capacitor fails when trying to load the request content. This happens during the fetch requests, and any subsequent `const data = await res.json();` calls are not executed.

I'm confused about why the `nativeResponse.headers` are undefined, as I expected fetch to work out of the box with `CapacitorHttp`. I'm not sure if I'm misusing fetch in combination with Capacitor, or if this is an Android-related issue. I've searched for similar issues and experimented with various implementation modifications, but I haven't directly used `CapacitorHttp.post`. I was hoping to avoid that, if possible. It feels like I'm very close to getting the whole package (cookies + request) working using the standard fetch, so any help or confirmation that this is not a bug would be appreciated.

### Capacitor setup

I made some modifications in the Android Manifest to allow text content requests:

```xml
 <application
	 ...
        android:usesCleartextTraffic="true">
	...
```

And my Capacitor config:

```js
...
  server: {
    //androidScheme: 'https'
  },
  plugins: {
    CapacitorHttp: {
      enabled: true,
    },
    CapacitorCookies: {
      enabled: true
    }
  },
  android: {
    allowMixedContent: true
  },
....
```

### Reproducing the issue

I believe the issue might be related to me misusing fetch or misunderstanding the HTTP plugin usage of Capacitor. If initial feedback suggests so, I can either move this to a discussion or create a reproduction repo (smaller than the one below). It might also be related to my server configuration, although I'm using fairly standard request headers and content on both sides.

The project I'm experiencing this issue with is [here](https://github.com/tbscode/tims-stack-anystack).

**Comment by tbscode** [comment]
Ok i've decided to poke a little i the capacitor code and actually the bug seems to be related to the captialization of `content-type` in the response headers.

If I change the [line](https://github.com/ionic-team/capacitor/blob/6843d9642fad9a322579cbe5f01563929a83dbf5/android/capacitor/src/main/assets/native-bridge.js#L387) to:

```js
    let data = !nativeResponse.headers['content-type'].startsWith('application/json')
```

All works fine, so I assume django is changing the naming of the header? This should be an issue with other servers to I assume?

**Comment by tbscode** [comment]
So this does seem to be a bug, the simple update `Content-Type` fixes all my issues everything works as expected now.

I'm not sure what causes the `content-type` header to be lowercase I assume this is prob django, though if this is an allowed shema it should prob also be supported right?

**Comment by jrobeson** [comment]
@tbscode indeed. it should work with `content-type` as well as `Content-Type`. There probably needs to be a wider look at all header usage to make sure all reads are case-insensitive.

**Comment by gerhardcit** [comment]
This issue makes HMR with on the phone completely unusable. When can we expect this to be fixed?

<img width="556" alt="Screenshot 2023-06-14 at 11 57 38" src="https://github.com/ionic-team/capacitor/assets/5508278/6431c409-25fe-4439-9d35-6ae3a3606257">

**Comment by tbscode** [comment]
So my current workaround is to apply `Option 2` on the fly.

I simply made a fork of capacitor and patched `native-bridge.js` as described above.

Then I apply the patch - by overwriting `native-bridge.js` via `postinstall` in my `package.json`:

```json
"postinstall": "curl -o node_modules/@capacitor/android/capacitor/src/main/assets/native-bridge.js https://raw.githubusercontent.com/tbscode/tims-capacitor/main/android/capacitor/src/main/assets/native-bridge.js"
```

This approach also works on CI since the post-install script automatically is executed by npm.

I've also tried to rebuild the android package with the fix described in `Option 2` though the build process is currently a little to involved for me.

**Comment by kjr-lh** [comment]
Just run into this, had to patch our version of capacitor even further (already using https://github.com/ionic-team/capacitor/pull/6206 )

Seems really weird that the java code is parsing the json only for the bridge code to stringify it again 🤷

**Comment by jcesarmobile** [comment]
should be fixed by https://github.com/ionic-team/capacitor/pull/6206

if people is still facing the issue, please, create a new issue and provide a sample app that reproduces the issue
